specification/quota/resource-manager/Microsoft.Quota/stable/2021-03-15/quota.json

```diff
@@ -0,0 +1,1215 @@
+            "schema": {
+              "$ref": "#/definitions/QuotaRequestSubmitResponse201"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ExceptionResponse"
+            }
+          }
+        }
+      }
+    },
+    "/{scope}/providers/Microsoft.Quota/quotaLimits": {
+      "get": {
+        "tags": [
+          "QuotaInformation"
+        ],
+        "description": "Get a list of current quota limits and usages of all resources. The response from this GET operation can be leveraged to submit requests to update a quota.",
+        "operationId": "Quota_List",
+        "x-ms-examples": {
+          "Quotas_listUsagesForCompute": {
+            "$ref": "./examples/getComputeUsages.json"
+          },
+          "Quotas_listUsagesMachineLearningServices": {
+            "$ref": "./examples/getMachineLearningServicesUsages.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/ScopeInPath"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. Quota information will be returned in the response with pagination.",
+            "schema": {
+              "$ref": "#/definitions/QuotaLimits"
+            },
+            "headers": {
+              "ETag": {
+                "description": "Current entity state version. Should be treated as opaque and used to make conditional HTTP requests.",
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ExceptionResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/{scope}/providers/Microsoft.Quota/quotaLimitsRequests/{id}": {
+      "get": {
+        "tags": [
+          "QuotaRequests"
+        ],
+        "x-ms-examples": {
+          "QuotaRequestStatus": {
+            "$ref": "./examples/getQuotaRequestStatusById.json"
+          },
+          "QuotaRequestInProgress": {
+            "$ref": "./examples/getQuotaRequestStatusInProgress.json"
+          },
+          "QuotaRequestFailed": {
+            "$ref": "./examples/getQuotaRequestStatusFailed.json"
+          }
+        },
+        "description": "Gets the quota request details and status by quota request ID for the resources of the resource provider at a specific location. The quota request ID **id** is returned in the response of the PUT operation.",
```

Comment:
>The quota request ID **id** is returned in the response of the PUT operation." [](http://example.com/codeflow?start=153&length=78)

There is patterns defined in the RPC to track aync operations (i.e. via Location and Azure-AsyncOperation header). Please follow the RPC.

[https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#asynchronous-operations](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#asynchronous-operations)

Reply:
Yes, we return  via Location and Azure-AsyncOperation header.
And ARM template and SDK works with it.
Is there a way to specify in the swagger?
The SDK generates the code to check it and ARM template deployment also checks it. It's included in the response header as per spec you specified.

Reply:
Updated response code to 202.
The headers are included as per https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#asynchronous-operations
 via Location and Azure-AsyncOperation header) and retry-after.
Do we need to specify in swagger?
The SDK and ARM template looks for those headers and works accordingly.

Reply:
Added the Async headers in examples.

Reply:
Thanks for adding the headers in the examples.

Please correct the description to indicate that the url for tracking is available in the headers not the ID in the body.